pinot-core/src/main/java/com/linkedin/pinot/core/segment/index/loader/IndexLoadingConfig.java

```diff
@@ -0,0 +1,195 @@
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.segment.index.loader;
+
+import com.linkedin.pinot.common.config.AbstractTableConfig;
+import com.linkedin.pinot.common.config.IndexingConfig;
+import com.linkedin.pinot.common.segment.ReadMode;
+import com.linkedin.pinot.core.data.manager.config.InstanceDataManagerConfig;
+import com.linkedin.pinot.core.indexsegment.generator.SegmentVersion;
+import com.linkedin.pinot.core.segment.index.loader.columnminmaxvalue.ColumnMinMaxValueGeneratorMode;
+import com.linkedin.pinot.core.startree.StarTreeFormatVersion;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+
+/**
+ * Table level index loading config.
+ */
+public class IndexLoadingConfig {
+  private static final int DEFAULT_REALTIME_AVG_MULTI_VALUE_COUNT = 2;
+
+  private ReadMode _readMode = ReadMode.DEFAULT_MODE;
+  private List<String> _sortedColumns = Collections.emptyList();
+  private Set<String> _invertedIndexColumns = new HashSet<>();
+  private Set<String> _noDictionaryColumns = new HashSet<>();
+  private SegmentVersion _segmentVersion = SegmentVersion.DEFAULT_VERSION;
+  private StarTreeFormatVersion _starTreeVersion = StarTreeFormatVersion.DEFAULT_VERSION;
+  private boolean _enableDefaultColumns = true;
+  private ColumnMinMaxValueGeneratorMode _columnMinMaxValueGeneratorMode = ColumnMinMaxValueGeneratorMode.DEFAULT_MODE;
+  private int _realtimeAvgMultiValueCount = DEFAULT_REALTIME_AVG_MULTI_VALUE_COUNT;
+
+  public IndexLoadingConfig(@Nullable InstanceDataManagerConfig instanceDataManagerConfig,
+      @Nullable AbstractTableConfig tableConfig) {
+    // Extract config from instance config
+    if (instanceDataManagerConfig != null) {
+      ReadMode instanceReadMode = instanceDataManagerConfig.getReadMode();
+      if (instanceReadMode != null) {
+        _readMode = instanceReadMode;
+      }
+
+      String instanceSegmentVersion = instanceDataManagerConfig.getSegmentFormatVersion();
+      if (instanceSegmentVersion != null) {
+        _segmentVersion = SegmentVersion.valueOf(instanceSegmentVersion.toLowerCase());
+      }
+
+      _enableDefaultColumns = instanceDataManagerConfig.isEnableDefaultColumns();
+
+      String avgMultiValueCount = instanceDataManagerConfig.getAvgMultiValueCount();
+      if (avgMultiValueCount != null) {
+        _realtimeAvgMultiValueCount = Integer.valueOf(avgMultiValueCount);
+      }
+    }
+
+    // Extract config from table indexing config
```

Comment:
IMO, extract might be more accurate because most configs does not exist in InstanceDataManagerConfig